files/zh-cn/web/javascript/reference/classes/public_class_fields/index.md

```diff
@@ -223,99 +166,61 @@ class DerivedWithConstructor extends Base {
   }
 }
 
-const instance2 = new DerivedWithConstructor(); // Logs 1
+const instance2 = new DerivedWithConstructor(); // 打印 1
 ```
 
-> **备注：** 在类字段规范最终确定使用 `[[Define]]` 语义前，包括 [Babel](https://babeljs.io/) 和 [tsc](https://www.typescriptlang.org/) 在内的大多数转译器都将类字段转换为 `DerivedWithConstructor` 的形式，这在类字段标准化后产生了一些错误。
-
-## 公有方法
-
-### 公有静态方法
+> **备注：** 在类字段规范最终确定为`[[DefineOwnProperty]]`语义之前，包括 [Babel](https://babeljs.io/) 和 [tsc](https://www.typescriptlang.org/)在内的大多数转译器都将类字段转换为 `DerivedWithConstructor` 形式，这在类字段规范化之后造成了一些微妙的错误。
 
-关键字 **`static`** 将为一个类定义一个静态方法。静态方法不是在一个实例之上被调用，而是在类自身之上被调用。它们通常是工具函数，比如用来创建或者复制对象。
-
-```js
-class ClassWithStaticMethod {
-  static staticMethod() {
-    return "static method has been called.";
-  }
-}
-
-console.log(ClassWithStaticMethod.staticMethod());
-// 预期输出值："static method has been called."
-```
-
-静态方法是在类的声明阶段用 {{jsxref("Global_Objects/Object/defineProperty", "Object.defineProperty()")}} 方法添加到类的构造函数中的。静态方法是可编辑、不可枚举和可配置的。
+## 示例
 
-### 公有实例方法
+### 使用类字段
 
-正如其名，公有实例方法是可以在类的实例中使用的。
+类字段不能依赖于构造函数的参数，因此字段初始化器通常会对每个实例评估相同的值（除非同一表达式每次评估的值都不同，例如 {{jsxref("Date.now()")}} 或对象初始化器）。
 
-```js
-class ClassWithPublicInstanceMethod {
-  publicMethod() {
-    return "hello world";
-  }
+```js example-bad
+class Person {
+  name = nameArg; // nameArg 在构造函数的作用域外
+  constructor(nameArg) {}
 }
-
-const instance = new ClassWithPublicInstanceMethod();
-console.log(instance.publicMethod());
-// 预期输出值："hello world"
 ```
 
-公有实例方法是在类的声明阶段用 {{jsxref("Global_Objects/Object/defineProperty", "Object.defineProperty()")}} 方法添加到类中的。静态方法是可编辑、不可枚举和可配置的。
-
-你可以使用生成器（generator）、异步和异步生成器方法。
-
-```js
-class ClassWithFancyMethods {
-  *generatorMethod() {}
-  async asyncMethod() {}
-  async *asyncGeneratorMethod() {}
+```js example-good
+class Person {
+  // Person 的所有实例会有相同的 name
+  name = "Dragomir";
 }
 ```
 
-在实例的方法中，`this` 指向的是实例本身，你可以使用 `super` 访问到父类的原型，由此你可以调用父类的方法。
+不过，即使声明一个空的类字段也是有好处的，因为这表明了字段的存在，从而允许类型检查程序和人类读者静态分析类的形状。
 
 ```js
-class BaseClass {
-  msg = "hello world";
-  basePublicMethod() {
-    return this.msg;
-  }
-}
-
-class SubClass extends BaseClass {
-  subPublicMethod() {
-    return super.basePublicMethod();
+class Person {
+  name;
+  age;
+  constructor(name, age) {
+    this.name = name;
+    this.age = age;
   }
 }
-
-const instance = new SubClass();
-console.log(instance.subPublicMethod());
-// 预期输出值："hello world"
 ```
 
-`getter` 和 `setter` 是和类的属性绑定的特殊方法，分别会在其绑定的属性被取值、赋值时调用。使用 [get](/zh-CN/docs/Web/JavaScript/Reference/Functions/get) 和 [set](/zh-CN/docs/Web/JavaScript/Reference/Functions/set) 语法定义实例的公有 `getter` 和 `setter`。
+上面的代码看似重复，但考虑一下 `this` 被动态更改的情况：明确的字段声明清楚地表明了哪些字段一定会出现在实例中。
 
```

Comment:
````suggestion

```js
class Person {
  name;
  age;
  constructor(properties) {
    Object.assign(this, properties);
  }
}
```

由于初始化器是在基类执行完成后求值的，因此你可以访问由基类构造函数创建的属性。

````